client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/pattern-list-panel.tsx

```diff
@@ -23,14 +29,20 @@ const PatternListPanel = ( {
 	label,
 	description,
 	onSelect,
+	recordTracksEvent,
 }: PatternListPanelProps ) => {
+	const translate = useTranslate();
+	const [ isShowMorePatterns, setIsShowMorePatterns ] = useState( false );
 	const categoryPatterns = selectedCategory ? patternsMapByCategory[ selectedCategory ] : [];
-
 	const category = useMemo(
 		() => selectedCategory && categories.find( ( { name } ) => name === selectedCategory ),
 		[ categories, selectedCategory ]
 	);
 
+	const hasNonPriorityPatterns = categoryPatterns?.find(
```

Comment:
[nit] I found we can just filter the priority patterns here and check the length before/after filtering to determine whether to show the `Show more patterns` button. Then, we don't need to pass `isShowMorePatterns` to `PatternSelector` and `PatternListRenderer` anymore. But I'm fine with the current way 🙂